CenterForOpenScience/osf.io
# [Feature] Make sync add ons sync on move [OSF-7201]

## Purpose
Moves of multiple files fail on github.


## Changes
Add a queue to sync move files instead of attempting to do them in parallel.


## Side effects
This does not just affect github, but also dataverse (see jira comment).



## Ticket
https://openscience.atlassian.net/browse/OSF-7201


[#OSF-7201]

## QA Considerations
1. Test github and dataverse (see Jira comment for PR that allows moves on dataverse).
2. Test that you can move single and multiple files.
3. Do some crazy dragging around and make sure no bad actions occur. You will definitely see 'please wait for action to finish' with this change, but that should prevent most bad behaviors from getting through
4. (This is from Brian G): I would be interested to know the consequences of bad and lost connections on this feature. Is it easy to recover from? For example, if I do a partial move, lose connection, then reload the page, is it easy to continue where it left off or are there going to be problems under some circumstances?
